app/sidekiq/terms_of_use/sign_up_service_updater_job.rb

```diff
@@ -48,12 +48,40 @@ def log_updated_icn
       end
     end
 
+    def client
+      @client ||= MAP::SignUp::Service.new
+    end
+
+    def status
+      @status ||= client.status(icn: mpi_profile.icn)
+    end
+
+    def declined?
+      status[:opt_out] == true
+    end
+
+    def accepted?
+      status[:agreement_signed] == true
+    end
+
+    def agreement_unchanged?
+      return false unless terms_of_use_agreement
+
+      unchanged = (terms_of_use_agreement.declined? == declined?) && (terms_of_use_agreement.accepted? == accepted?)
```

Comment:
In fact, I was playing around with seeing how simplified we can make this function, what do you think of this?

>     def agreement_unchanged?
>      if terms_of_use_agreement.declined? != status[:opt_out] ||
>        terms_of_use_agreement.accepted? != status[:agreement_signed]
>        return false
>      end
>
>      Rails.logger.info("#{LOG_TITLE} Agreement not changed", { icn: user_account.icn })
>      true
>     end

Reply:
I'm not 100% on this, because I think if ToU is `nil` the `if` block would break on `declined?`.  We likely would still need a `return false unless terms_of_use_agreement` at the very least right? @bosawt

Reply:
Ya when we initially looked at it I think we thought that would be a good idea, but then I looked again at the rest of the file and basically a ton of things will break if there isn't a `terms_of_use_agreement`, so I wasn't sure anymore if that's necessary in this function